File: Previous Contest/3487. Maximum Unique Subarray Sum After Deletion/Solution.java
import java.util.*;
import java.lang.Math;

class Solution {
    public int maxSum(int[] nums) {
        // N: len(nums)
        // Time Complexity: O(N)
        // Space Complexity: O(N)
        int n = nums.length;
        int summ = 0;
        int maxVal = Integer.MIN_VALUE;
        boolean zeroFlag = false;
        HashSet<Integer> visited = new HashSet<>();
        for (int i = 0; i < n; i++) {
            maxVal = Math.max(maxVal, nums[i]);
            
            if (nums[i] < 0 || visited.contains(nums[i])) continue;
            if (nums[i] == 0) zeroFlag = true;

            visited.add(nums[i]);
            summ += nums[i];
        }
        return (summ == 0 && zeroFlag == false) ? maxVal : summ;
    }
}

File: Previous Contest/3487. Maximum Unique Subarray Sum After Deletion/Solution.py
from typing import List

class Solution:
    def maxSum(self, nums: List[int]) -> int:
        # N: len(nums)
        # Time Complexity: O(N)
        # Space Complexity: O(N)
        n = len(nums)
        visited = set()
        summ = 0
        maxVal = float('-inf')
        zeroFlag = False
        for i in range(n):
            maxVal = max(maxVal, nums[i])

            if nums[i] < 0 or nums[i] in visited:
                continue
            if nums[i] == 0:
                zeroFlag = True

            visited.add(nums[i])
            summ += nums[i]

        return maxVal if (not zeroFlag and summ == 0) else summ